decidim-admin/app/commands/decidim/admin/update_component.rb

```diff
@@ -57,6 +59,28 @@ def settings_changed?
       def current_settings
         @component.attributes["settings"]
       end
+
+      # Keep previous values for disabled settings
+      def restore_disabled_settings!
+        browse_disabled_settings("global") do |attribute|
+          form.settings[attribute] = @previous_settings.dig("global", attribute)
+        end
+
+        browse_disabled_settings("step") do |attribute|
+          form.default_step_settings[attribute] = @previous_settings.dig("default_step", attribute) if form.default_step_settings.present?
+          if form.step_settings.present?
+            form.step_settings.each do |step_name, step|
+              step[attribute] = @previous_settings.dig("steps", step_name, attribute)
+            end
+          end
+        end
+      end
+
+      def browse_disabled_settings(settings_name)
+        @component.manifest.settings(settings_name).attributes
+                  .select { |_attribute, obj| obj.disabled?(component: @component) }
```

Comment:
Ok, I think that is definitely a bad idea to have a `disabled` option when a lot of settings are called `...._enabled`. It seems that it could affect the value, while it only affects the ability of admins to modify it. If you are ok with the `readonly` name I'll change it that way.